R/pivot_wider.R

```diff
@@ -96,9 +91,16 @@ pivot_wider..tidytable <- function(.df,
     glue_df <- distinct.(.df, !!!syms(names_from))
     values_from_reps <- nrow(glue_df)
     glue_df <- vec_rep(glue_df, length(values_from))
-    glue_df$.value <- vec_rep_each(values_from, values_from_reps)
+    glue_df <- mutate.(glue_df,
+                       .value = vec_rep_each(values_from, values_from_reps),
+                       .before = 1)
 
     glue_vars <- as.character(glue_data(glue_df, names_glue))
+    # mimic column names assigned by data.table::dcast()
+    if (length(values_from) <= 1) {
+      glue_df[[".value"]] <- NULL
```

Comment:
Can you use one of these options instead of `glue_df[[".value"]] <- NULL`?

They both basically do the same thing but use the `data.table` way to remove columns. (Not that it matters here but they both avoid modify-by-reference as well.)

Option 1: `glue_df <- dt_j(glue_df, .value := NULL)`

Option 2: `glue_df <- mutate.(glue_df, .value = NULL)`